airbyte-integrations/connectors/destination-teradata/src/test-integration/java/io/airbyte/integrations/destination/teradata/TeradataDestinationSSLAcceptanceTest.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.teradata;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.map.MoreMaps;
+import io.airbyte.commons.string.Strings;
+import io.airbyte.db.factory.DataSourceFactory;
+import io.airbyte.db.jdbc.DefaultJdbcDatabase;
+import io.airbyte.db.jdbc.JdbcDatabase;
+import io.airbyte.db.jdbc.JdbcSourceOperations;
+import io.airbyte.db.jdbc.JdbcUtils;
+import io.airbyte.integrations.base.AirbyteTraceMessageUtility;
+import io.airbyte.integrations.base.JavaBaseConstants;
+import io.airbyte.integrations.destination.ExtendedNameTransformer;
+import io.airbyte.integrations.standardtest.destination.JdbcDestinationAcceptanceTest;
+import io.airbyte.protocol.models.v0.AirbyteConnectionStatus;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.sql.SQLException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import javax.sql.DataSource;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TeradataDestinationSSLAcceptanceTest extends JdbcDestinationAcceptanceTest {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(TeradataDestinationAcceptanceTest.class);
+  private final ExtendedNameTransformer namingResolver = new ExtendedNameTransformer();
+
+  private JsonNode configJson;
+  private JdbcDatabase database;
+  private DataSource dataSource;
+  private TeradataDestination destination = new TeradataDestination();
+  private final JdbcSourceOperations sourceOperations = JdbcUtils.getDefaultSourceOperations();
+
+  @Override
+  protected String getImageName() {
+    return "airbyte/destination-teradata:dev";
+  }
+
+  @Override
+  protected JsonNode getConfig() {
+    return configJson;
+  }
+
+  public JsonNode getStaticConfig() throws Exception {
+    final JsonNode config = Jsons.deserialize(Files.readString(Paths.get("secrets/sslconfig.json")));
+    ((ObjectNode) config).put("ssl", true);
```

Comment:
why not have `ssl` and `sslmode` specified in `secrets/sslconfig.json` instead of adding them programmatically.

Reply:
also, `spec.json` expects `ssl_mode` but here you are adding `sslmode`

Reply:
Removed providing ssl options from this class and will provide latest config json file.

Reply:
Yes, spec.json expects ssl_mode. This testcase was got succeeded when ssl is true, if no ssl_mode specified with any value like require, prefer.., it is taking require as default value. 
![image](https://user-images.githubusercontent.com/23031170/214711346-d1969e0e-a59f-477a-99f1-f362a9c0cb5a.png)